src/vehicle.cpp

```diff
@@ -4693,7 +4794,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+
+            if( is_body_collision && parts[ret.part].enabled &&
```

Comment:
Enabled could be here, but `damage_from_chainsaw` should check whether the part is a chainsaw anyway, in case someone calls it for something else.

Later on, if you add more chainsaw types, `damage_from_chainsaw` will use their values rather than hardcoded 70. Then those parts will have to be chainsaws, which would means that the chainsaw flag check should be there.